src/MissionManager/StructureScanComplexItem.cc

```diff
@@ -519,6 +513,9 @@ void StructureScanComplexItem::_rebuildFlightPolygon(void)
     } else {
         _entryVertex = savedEntryVertex;
     }
+
+    _recalcScanDistance();
```

Comment:
This will calc recalc the scan distance at times when it is not needed since the number of camera shots can change without the scan distance being affected. The signal connected to the flight path polygon is the better way to do it. Since the two cases which affect scan distance are polygon change or layer count change.